org-code-javabuilder/theater/src/main/java/org/code/theater/GifWriter.java

```diff
@@ -48,6 +50,13 @@ public GifWriter createGifWriter(ByteArrayOutputStream out) {
    */
   public void writeToGif(BufferedImage img, int delay) {
     try {
+      if (this.imageOutputStream.length() > MAX_STREAM_LENGTH_BYTES) {
+        Theater.stage.close();
```

Comment:
It does seem a bit weird for GifWriter to call Theater.stage directly, but totally fine with this for now. I think once we have a cleanup notifier in place, we can hook into that as well, but this is fine for now.

Reply:
yeah I'll add a comment to clean this up once we have a notifier